### PR TITLE
[RELEASE] 8.0.0

### PR DESCRIPTION
**Pull request by mdybich**
**Summary**

- upgrading to Angular 8
- `InfiniteScrollEvent` interface is renamed to `IInfiniteScrollEvent` (it follows the TSLint rule - `interface name must start with a capitalized I(interface-name)`)

### Requirements
Make sure these boxes are checked:  
- [x] tests are updated (added/changed/removed)  
- [x] tests run successfully (local) 
- [x] tests run successfully (TravisCI) 
- [x] please follow this [styleguide](https://gist.github.com/stephenparish/9941e89d80e2bc58a153#subject-line) for this pr title

### CODE REVIEW

File: examples/package.json [hunk 1]
```diff
@@ -12,26 +12,25 @@
...
     "rxjs-compat": "^6.0.0-rc.0",
-    "zone.js": "^0.8.26"
+    "zone.js": "0.9.1"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "~0.6.3",
-    "@angular/cli": "^7.0.0",
-    "@angular/compiler-cli": "^7.0.0",
-    "@angular/language-service": "^7.0.0",
+    "@angular-devkit/build-angular": "0.801.2",
```
Comment (orizens) [inline examples/package.json]: are you sure 0.801.2 is the correct version?

File: examples/package.json [hunk 2]
```diff
@@ -12,26 +12,25 @@
...
+    "@angular/common": "^8.0.0",
+    "@angular/compiler": "^8.0.0",
+    "@angular/core": "^8.0.0",
+    "@angular/forms": "^8.0.0",
+    "@angular/platform-browser": "^8.0.0",
+    "@angular/platform-browser-dynamic": "^8.0.0",
+    "@angular/router": "^8.0.0",
     "core-js": "^2.4.1",
-    "ngx-infinite-scroll": "file:../ngx-infinite-scroll-7.0.1.tgz",
+    "ngx-infinite-scroll": "file:../ngx-infinite-scroll-8.0.0.tgz",
```
Comment (orizens) [inline examples/package.json]: this can be removed completely.
thanks.

File: examples/package.json [hunk 3]
```diff
@@ -47,6 +46,6 @@
     "protractor": "~5.1.2",
     "ts-node": "~3.2.0",
     "tslint": "~5.7.0",
-    "typescript": "3.1.1"
+    "typescript": "3.4.5"
```
Comment (orizens) [inline examples/package.json]: we can use here ``^3.1.1``

File: package.json [hunk 4]
```diff
@@ -69,12 +69,12 @@
     "ts-helpers": "1.1.2",
     "ts-loader": "2.0.3",
     "tsickle": "^0.34.0",
-    "tslint": "4.5.1",
-    "typescript": "3.1.6",
+    "tslint": "5.18.0",
+    "typescript": "3.4.5",
```
Comment (orizens) [inline package.json]: we can use here ^3.1.1

File: examples/package.json (hunk 1 above)
Comment (mdybich) [reply examples/package.json]: Yeah, it is correct version:
https://www.npmjs.com/package/@angular-devkit/build-angular
https://github.com/angular/angular-cli#tools

File: examples/package.json (hunk 2 above)
Comment (mdybich) [reply examples/package.json]: Fixed in https://github.com/orizens/ngx-infinite-scroll/pull/333/commits/ad0b645d37f7f61e8353e036cd864b6b5e759e04

File: examples/package.json (hunk 3 above)
Comment (mdybich) [reply examples/package.json]: Fixed in https://github.com/orizens/ngx-infinite-scroll/pull/333/commits/ad0b645d37f7f61e8353e036cd864b6b5e759e04

File: package.json (hunk 4 above)
Comment (mdybich) [reply package.json]: Fixed in https://github.com/orizens/ngx-infinite-scroll/pull/333/commits/ad0b645d37f7f61e8353e036cd864b6b5e759e04

**Comment by mdybich** [comment]
@orizens all comments are fixed. I believe that PR should be ready to merge now.

**Comment by orizens** [comment]
@mdybich thanks for this pr.
version 8 will be published soon to npm registry.

**Pull request merged**